### PR TITLE
bpo-46587: Skip tests if strftime does not support glibc extension (GH-31873)

### PR DESCRIPTION
**Pull request by tiran**
https://bugs.python.org/issue46587

### CODE REVIEW

File: Lib/test/support/__init__.py [hunk 1]
```diff
@@ -520,6 +520,8 @@ def requires_subprocess():
     """Used for subprocess, os.spawn calls, fd inheritance"""
     return unittest.skipUnless(has_subprocess_support, "requires subprocess support")
 
+# Does strftime() support glibc extension like '%4Y'?
+has_strftime_extension = time.strftime("%4Y") != "%4Y"
```
Comment (vstinner) [inline Lib/test/support/__init__.py]: Would plural be more appropriate if there is more than one extension?

`man strftime` says:

> Glibc notes: Glibc provides some **extensions** for  conversion  specifications.

File: Misc/NEWS.d/next/Tests/2022-03-14-17-10-35.bpo-46587.ASDsJX.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Skip tests if platform's ``strftime`` does not support non-portal glibc
```
Comment (vstinner) [inline Misc/NEWS.d/next/Tests/2022-03-14-17-10-35.bpo-46587.ASDsJX.rst]: typo?

```suggestion
Skip tests if platform's ``strftime`` does not support non-portable glibc
```

**Review by vstinner — APPROVED** [review]
LGTM.

**Comment by sweeneyde** [comment]
This may have caused some trouble on Windows builds. See https://bugs.python.org/issue47037

**Pull request merged**